plugins/module_utils/network/nxos/config/interfaces/interfaces.py

```diff
@@ -433,8 +431,11 @@ def add_commands(self, d, obj_in_have=None):
         if "duplex" in d:
             commands.append("duplex " + d["duplex"])
         if "enabled" in d:
-            have_enabled = obj_in_have.get(
-                "enabled", self.default_enabled(d, obj_in_have)
+            have_enabled = (
+                obj_in_have.get(
+                    "enabled", self.default_enabled(d, obj_in_have)
+                )
+                or False
```

Comment:
I would think this would work with subinterfaces as well, but I don't see that in your test below.

Reply:
Added now.